test/parallel/test-assert-async.js

```diff
@@ -10,13 +10,13 @@ const wait = promisify(setTimeout);
 // expected output and by verifying that they do not work sync
 
 assert.rejects(
-  () => assert.fail(),
+  async () => assert.fail(),
   common.expectsError({
     code: 'ERR_ASSERTION',
     type: assert.AssertionError,
     message: 'Failed'
   })
-);
+).then(common.mustCall());
```

Comment:
Nit: this test should now be redundant if I am not mistaken.

Reply:
I'm assuming you mean that this test is redundant with the other test [here](https://github.com/nodejs/node/blob/edeb64caa5726433828cff76e154afdba4a0b5f8/test/parallel/test-assert-async.js#L73-L84). However, this test is a bit different from the other one because it throws asynchronously, and so the promise returned by `assert.rejects` fulfills (whereas in the other test, the function throws synchronously and the promise returned by `assert.rejects` rejects and needs to be caught separately).

Reply:
Actually, I just realized what you meant: this test is now redundant with [this other test](https://github.com/nodejs/node/blob/edeb64caa5726433828cff76e154afdba4a0b5f8/test/parallel/test-assert-async.js#L26-L36). I agree, I'll remove the other one.